src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/ApprenticesFilterModel.cs

```diff
@@ -0,0 +1,173 @@
+
+                    if (PageNumber > totalPages - 2)
+                        pageNumberSeed = totalPages - 4;
+                }
+
+                for (var i = 0; i < totalPageLinks; i++)
+                {
+                    var link = new PageLink
+                    {
+                        Label = (pageNumberSeed + i).ToString(),
+                        AriaLabel = $"Page {pageNumberSeed + i}",
+                        IsCurrent = pageNumberSeed + i == PageNumber? true : (bool?)null,
+                        RouteData = BuildPagedRouteData(pageNumberSeed + i)
+                    };
+                    links.Add(link);
+                }
+
+                //next link
+                if (totalPages > 1 && PageNumber < totalPages)
+                {
+                    links.Add(new PageLink
+                    {
+                        Label = "Next",
+                        AriaLabel = "Next page",
+                        RouteData = BuildPagedRouteData(PageNumber + 1)
+                    });
+                }
+
+                return links;
+            }
+        }
+
+        public Dictionary<string, string> RouteData => BuildRouteData();
+
+        private Dictionary<string, string> BuildRouteData()
+        {
+            var routeData = new Dictionary<string, string>();
+
+            if (!string.IsNullOrWhiteSpace(SearchTerm))
+            {
+                routeData.Add(nameof(SearchTerm), SearchTerm);
+            }
+
+            if (!string.IsNullOrWhiteSpace(SelectedProvider))
+            {
+                routeData.Add(nameof(SelectedProvider), SelectedProvider);
+            }
+
+            if (!string.IsNullOrWhiteSpace(SelectedCourse))
+            {
+                routeData.Add(nameof(SelectedCourse), SelectedCourse);
+            }
+
+            if (SelectedStatus.HasValue)
+            {
+                routeData.Add(nameof(SelectedStatus), SelectedStatus.Value.ToString());
+            }
+            
+            if (SelectedEndDate.HasValue)
+            {
+                routeData.Add(nameof(SelectedEndDate), SelectedEndDate.Value.ToString("yyyy-MM-dd"));
+            }
+
+            return routeData;
+        }
+        
+        private Dictionary<string, string> BuildPagedRouteData(int pageNumber)
+        {
+            var routeData = BuildRouteData();
+            
+            routeData.Add(nameof(PageNumber), pageNumber.ToString());
+
+            if (!string.IsNullOrEmpty(SortField))
+            {
+                routeData.Add(nameof(SortField), SortField);
+
+                routeData.Add(nameof(ReverseSort), ReverseSort.ToString());
+            }
+
+            return routeData;
+        }
+
+        public Dictionary<string, string> BuildSortRouteData(string sortField)
+        {
+            var routeData = BuildRouteData();
+
+            var reverseSort = !string.IsNullOrEmpty(SortField) 
+                              && SortField.ToLower() == sortField.ToLower() 
```

Comment:
It might be beneficial to name the parameter sortField into something different to make it easier to distinguish between the Property. It would just make it slightly easier to read code wise